### PR TITLE
Use the config generator to produce most of our job configs.

### PR DESCRIPTION
**Pull request by geeknoid**
- Extend the config generator to understand default resource values,
which avoids repeating them over and over.

### CODE REVIEW

File: prow/config/jobs/istio.io.yaml
Comment (howardjohn) [inline prow/config/jobs/istio.io.yaml:7]: not 1.3 as well? because of the different branching approach I assume?

Comment (geeknoid) [reply prow/config/jobs/istio.io.yaml:7]: Correct, there is no 1.3 branch at the moment.

**Comment by howardjohn** [comment]
Can a prow expert confirm that in order to rename a test job (as we are doing here, to remove duplicate names: https://github.com/istio/test-infra/issues/1402), we just need to switch the names and things should work? History for the jobs will be lost, but we can copy the test logs/results over in GCS manually and get them back?

LGTM pending verification on the above ^

File: prow/cluster/jobs/istio/istio/istio.istio.master.yaml
Comment (howardjohn) [inline prow/cluster/jobs/istio/istio/istio.istio.master.yaml]: can you modify these to not have postsubmit in the original name? I think there is the same for presubmit

Comment (geeknoid) [reply prow/cluster/jobs/istio/istio/istio.istio.master.yaml]: Done.

**Comment by geeknoid** [comment]
@howardjohn For the sake of shorter names, how about we tack on -postsubmit for post submit tests, but nothing for presubmit tests (which are the ones that show up in the GitHub UI)

WDYT?

**Comment by howardjohn** [comment]
Yes good idea

**Comment by howardjohn** [comment]
we could also prefix the repo name? So I specify `test` and it becomes
`istio-test-postsubmit-master`. I am neutral about that though, not sure if
its helpful or too much magic

**Comment by geeknoid** [comment]
OK, I've removed "presubmit", but left "postsubmit".

As for tacking on the repo name, that's a good idea IMO. But it'll require going through all the job names and removing their existing repo prefixes, and in the case of the istio/istio jobs, it'll require picking new naming schemes since the generated names end up being too long (limit is 63 chars). So I've left this for later.

**Comment by howardjohn** [comment]
If we are going to change job names, which will require some manual
efforts, it may be better to do it all at once.

**Comment by geeknoid** [comment]
@howardjohn PTAL.

**Comment by howardjohn** [comment]
lgtm!

@clarketm can you advise us on the best way to do a big change of job names like this?

> Can a prow expert confirm that in order to rename a test job (as we are doing here, to remove duplicate names: #1402), we just need to switch the names and things should work? History for the jobs will be lost, but we can copy the test logs/results over in GCS manually and get them back?

**Comment by clarketm** [comment]
> @clarketm can you advise us on the best way to do a big change of job names like this?
> 
> > Can a prow expert confirm that in order to rename a test job (as we are doing here, to remove duplicate names: #1402), we just need to switch the names and things should work? History for the jobs will be lost, but we can copy the test logs/results over in GCS manually and get them back?

@fejta @cjwagner - can one of you confirm that the assumption made by @howardjohn here is correct regarding job renaming and history preservation?

**Review by fejta — COMMENTED** [review]
Can we put the code changes in their own commit and the generated code change in a separate commit? I'm struggling to find what I'm supposed to review here.

Also, can we get a https://github.com/kubernetes/kubernetes/blob/master/.generated_files so that these PRs are easier to review? This will cause github to default to collapsing diffs in these files (you can still expand if you want to see them).

**Comment by fejta** [comment]
> Can a prow expert confirm that in order to rename a test job (as we are doing here, to remove duplicate names: #1402), we just need to switch the names and things should work?

Yes, with some caveats.

In addition to copying GCS files if you want to preserve history, you'll also need to use the status reconciler to ensure any status contexts get migrated (unless you're explicitly setting the context, which you should not do)

https://github.com/kubernetes/test-infra/tree/master/prow/cmd/status-reconciler

**Comment by howardjohn** [comment]
> .generated_files 

That seems to be for a kubernetes tool not github? But it doesn't seem like we can do it with https://help.github.com/en/articles/customizing-how-changed-files-appear-on-github? Which would be a great improvement

> In addition to copying GCS files if you want to preserve history, you'll also need to use the status reconciler to ensure any status contexts get migrated (unless you're explicitly setting the context, which you should not do)

I believe we have status reconciler set up, as we previous change the status contexts (to no longer set them explicitly), so seems like this should be good to go

**Comment by fejta** [comment]
Yes sorry I meant https://github.com/kubernetes/kubernetes/blob/master/.gitattributes

**Comment by geeknoid** [comment]
Generated yaml files now have the .gen.yaml suffix, and a .gitattributes file now excludes them.

PTAL.

**Review by fejta — APPROVED** [review]
Pretty nifty!
/hold
merge at your leisure

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
I would recommend we don't merge this until after 1.3 is out? Last time we did a status reconcile it was pretty slow, and I don't want to block any P0 changes going out

@fejta I recall last time we found the reconciliation was artificially rate limited, can we speed it up?

**Comment by fejta** [comment]
Not safely, no. Otherwise it will essentially cause a prow outage when it runs out of tokens. Was ghproxy being used last time?

Perhaps try changing job contexts more gradually?

Github provides a limited number of tokens per time period, and migrating each token consumes a few tokens, at least three: get old context, push new context, override old context. And github requires this to be done one context on one commit at a time.

**Comment by howardjohn** [comment]
> Not safely, no. Otherwise it will essentially cause a prow outage when it runs out of tokens. Was ghproxy being used last time?

I am not sure if we used ghproxy or not. It seems we don't have token usage stats on https://velodrome.istio.io so not sure if we are anywhere near limits or not anyways. Ideally we could have this run at night (or over the 3 day weekend.. oops) or something and just use elevated amount of tokens.

If not, for now we could revert the changes to istio/istio for now, and roll out gradually? That works for me, we will just need to disable the generate config check?

**Comment by howardjohn** [comment]
This can be done later, but opened https://github.com/istio/test-infra/issues/1762 to improve job UX further.

current config looks great, we agreed offline to merge friday after 1.3 is built

**Review by fejta — APPROVED** [review]

**Pull request merged**
